includes/functions.php

```diff
@@ -81,6 +81,11 @@ function acft_enqueue_google_fonts_file() {
     
     global $post;
     
+    // anticipate 404 errors or non-object queries where there is no $post->ID, just return.
+    if ( is_404() || !is_object($post) ) {
```

Comment:
I haven't tested the code yet but I think not enqueuing the Google fonts on 404 page could create issues for those users who have used ACF Typography in ACF Options. An example would be using the same header and footer as the rest of the site. We will likely need to do it differently. What are your thoughts?

Reply:
Yeah, I get that.

Don't merge this one. I've got it covered in a MAJOR pull request coming within the next day or two. It's everything I mentioned in this topic:
https://wordpress.org/support/topic/enqueue-fonts-styles-in-custom-function/

In order to accomplish those things, I ended up breaking your function for enqueue-ing Google fonts in half, using the first half as a new function, and it's called to enqueue Google fonts, and called by my new shortcode for inserting font stylesheets links/styles in-line.

Part of that function separation and code changes I did should resolve this. It now follow this logic:

- Function now allows for an optional $post_id to be supplied.
- If no $post_id supplied, it'll try to grab it from **inside** the loop with global $post (what you originally had).
- If still no $post_id, it'll try to grab it from **outside** the loop with global $wp_query.
- If there's still no $post_id after all of that, it will only check for option fields, and return them as the "all_options" variable without any array merging. 
- If a $post_id is present, it'll process post fields, block fields, option fields, and merge the arrays, and return the merged array as the "all_options" variable (what you originally had).

Give me another day or two to finish up the edits. Not sure if it should be released as v3.3 or v4.0, but I'll leave that up to you.